### ISSUE TITLE
HLSL: seems the register number of a cbuffer cannot be captured

### ISSUE THREAD

**Issue by StarsX** [KhronosGroup/glslang#496]
Hi,
I find another problem on HLSL cbuffer. It seems that the register number of a cbuffer cannot be captured for the corresponding struct declaration list.
As I wrote the following cbuffer declaration:

	cbuffer cbMatrices : register(b0)
	{
		float4x4 g_mWorldViewProj;
		float4x4 g_mWorld;
		float3x3 g_mWorldIT;
	};

the result SPIR-V code (related part):

	Decorate 13 Block
	Decorate 15 DescriptorSet 0

There seems a binding statement missing:

	Decorate 15 Binding 0

Although it is runnable on NVIDIA card and it seems that NVIDIA makes it binding 0 as default, it will crash on other cards. Anyway, I think it is necessary to generate this statement explicitly according to the spec.

Nevertheless, if the cbuffer is declared like this:

	cbuffer cbMatrices
	{
		float4x4 mWorldViewProj;
		float4x4 mWorld;
		float3x3 mWorldIT;
	} g_cbMatrices : register(b0);
	#define g_mWorldViewProj	g_cbMatrices.mWorldViewProj
	#define g_mWorld			g_cbMatrices.mWorld
	#define g_mWorldIT			g_cbMatrices.mWorldIT
it produces

	Decorate 15(g_cbMatrices) DescriptorSet 0
	Decorate 15(g_cbMatrices) Binding 0
However, this syntax is invalid in HLSL, which cannot be compiled by Microsoft FXC.

Finally, I found that, in HlslGrammar::acceptStruct(TType& type), the result of the register parsing by acceptPostDecls(type) does not passed to the qualifier of the struct declaration list. Thus, after I simply added one statement to the code (#line 1343 hlslGrammar.cpp) as:

	else
	{
		qualifier.layoutBinding = type.getQualifier().layoutBinding;
		new(&type) TType(typeList, structName, qualifier); // sets EbtBlock
	}
the register can be captured and passed to the cbuffer type list successfully, and it generates SPIR-V code "Decorate 15 Binding 0".

Thanks for your attention.

**Comment by johnkslang** [comment]
Thanks for the input; it's now time to finish sorting out register numbers, etc.

In these two places:

```
cbuffer cbMatrices : register(b0); // place 1
{
    float4x4 mWorldViewProj; ...
} g_cbMatrices : register(b0);     // place 2 (hypothetically, not suggesting to do this)
```

I would (by default) think `place 1` is modifying the type and `place 2` is modifying the instance. I'll proceed on that assumption that what is wrong is the instance is not inheriting everything it should from the type.

**Comment by johnkslang** [comment]
This is fixed by b804de6.
